cue/ui/gen.cue

```diff
@@ -93,3 +125,52 @@ VizLegendOptions: {
 VizTooltipOptions: {
 	mode: TooltipDisplayMode
 } @cuetsy(targetType="interface")
+// TODO copy back to appropriate place
+SingleStatBaseOptions: {
+	OptionsWithTextFormatting
+	reduceOptions: ReduceDataOptions
+	orientation: VizOrientation
+} @cuetsy(targetType="interface")
+// TODO copy back to appropriate place
+ReduceDataOptions: {
+	// If true show each row value
+	values?: bool
+	// if showing all values limit
+	limit?: number
+	// When !values, pick one value for the whole field
+	calcs: [...string]
+	// Which fields to show.  By default this is only numeric fields
+	fields?: string
+} @cuetsy(targetType="interface")
+// TODO copy back to appropriate place
+VizOrientation: "auto" | "vertical" | "horizontal" @cuetsy(targetType="enum")
+// TODO copy back to appropriate place
+OptionsWithTooltip: {
+	tooltip?: VizTooltipOptions
+} @cuetsy(targetType="interface")
+// TODO copy back to appropriate place
+OptionsWithLegend: {
+	legend?: VizLegendOptions
```

Comment:
This is a good example where we should discuss what undefined/optional means.

I think we need to put in a default value for this and tooltip mode rather than have them optional

Reply:
I generally prefer the required-with-default approach over having it be optional, as well